### PR TITLE
Feat/mongo enterprise

### PR DESCRIPTION
**Pull request by cristiGuranIonos**
## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: docs/resources/dbaas_mongo_cluster.md [hunk 1]
```diff
@@ -56,24 +106,33 @@ resource "random_password" "cluster_password" {
...
 * `location` - (Computed)[string] The connection string for your cluster. Updates to the value of the field force the cluster to be re-created.
 * `connections` - (Required)[string] Details about the network connection for your cluster. Updates to the value of the field force the cluster to be re-created.
     * `datacenter_id` - (Required)[true] The datacenter to connect your cluster to.
     * `lan_id` - (Required)[true] The LAN to connect your cluster to.
     * `cidr` - (Required)[true] The IP and subnet for the database. Must be same number as instances. Note the following unavailable IP ranges: 10.233.64.0/18, 10.233.0.0/18, 10.233.114.0/24. Please input in the correct format like IP/Subnet, exp: 192.168.10.0/24. See [Private IPs](https://www.ionos.com/help/server-cloud-infrastructure/private-network/private-ip-address-ranges/) and [Cluster Setup - Preparing the network](https://docs.ionos.com/reference/product-information/api-automation-guides/database-as-a-service/create-a-database#preparing-the-network).
+    * `whitelist` - (Optional)[true] List of whitelisted CIDRs
```
Comment (mimihalescu) [inline docs/resources/dbaas_mongo_cluster.md:119]: This is not implemented yet, right?

**Review by mimihalescu — APPROVED** [review]

File: docs/data-sources/dbaas_mongo_cluster.md [hunk 2]
```diff
@@ -47,7 +48,17 @@ The following attributes are returned by the datasource:
...
     * `time` 
     * `day_of_the_week`
 * `connection_string` - The physical location where the cluster will be created. This will be where all of your instances live. Updates to the value of the field force the cluster to be re-created. Available locations: de/txl, gb/lhr, es/vit"
+* `ram`          - The amount of memory per instance in megabytes. Required for enterprise edition.
+* `storage_size` - The amount of storage per instance in MB. Required for enterprise edition.
+* `storage_type` - The storage type used in your cluster. Required for enterprise edition.
+* `cores`        - The number of CPU cores per replica. Required for enterprise edition.
+* `backup` 
+  * `location`: The location where the cluster backups will be stored. If not set, the backup is stored in the nearest location of the cluster. Possible values are de, eu-south-2, or eu-central-2.
+* `from_backup` - The unique ID of the backup you want to restore. This attribute is immutable(disallowed in update requests).
```
Comment (avirtopeanu-ionos) [inline docs/data-sources/dbaas_mongo_cluster.md:62]: i tried testing this and

```
400 Bad Request: {"httpStatus":400,"messages":[{"message":"Payload parse error: Unknown field: fromBackup.","errorCode":"dbaas-err-api-01"}]}
```

File: docs/resources/dbaas_mongo_cluster.md (hunk 1 above)
Comment (cristiGuranIonos) [reply docs/resources/dbaas_mongo_cluster.md:119]: good catch!

File: docs/data-sources/dbaas_mongo_cluster.md (hunk 2 above)
Comment (cristiGuranIonos) [reply docs/data-sources/dbaas_mongo_cluster.md]: good find, removed

**Pull request merged**